### PR TITLE
Record the results of pod eviction by kubelet and controller-manager

### PR DESCRIPTION
**Pull request by jiaxuanzhou**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes ##56272

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/controller/util/node/controller_utils.go [hunk 1]
```diff
@@ -97,6 +97,7 @@ func DeletePods(kubeClient clientset.Interface, recorder record.EventRecorder, n
 		if err := kubeClient.CoreV1().Pods(pod.Namespace).Delete(pod.Name, nil); err != nil {
 			return false, err
 		}
+		recorder.Eventf(&pod, v1.EventTypeNormal, "NodeControllerEviction", "Successfully delete Pod %s from Node %s", pod.Name, nodeName)
```
Comment (yastij) [inline pkg/controller/util/node/controller_utils.go]: Use the past, deleted instead of delete

File: pkg/kubelet/eviction/eviction_manager.go [hunk 2]
```diff
@@ -576,18 +576,21 @@ func (m *managerImpl) evictPod(pod *v1.Pod, resourceName v1.ResourceName, evictM
...
-		Message: fmt.Sprintf(message, resourceName),
+		Message: msg,
 		Reason:  reason,
 	}
 	// record that we are evicting the pod
 	m.recorder.Eventf(pod, v1.EventTypeWarning, reason, evictMsg)
 	gracePeriod := int64(0)
 	err := m.killPodFunc(pod, status, &gracePeriod)
 	if err != nil {
+		m.recorder.Eventf(pod, v1.EventTypeWarning, "KubeletEviction", "failed to delete pod,Err %v", err)
```
Comment (yastij) [inline pkg/kubelet/eviction/eviction_manager.go]: We should limit emitting of events, from event-style guide, we shouldn’t emit events such as `failed to X`

File: pkg/kubelet/eviction/eviction_manager.go [hunk 3]
```diff
@@ -576,18 +576,21 @@ func (m *managerImpl) evictPod(pod *v1.Pod, resourceName v1.ResourceName, evictM
...
 		glog.Errorf("eviction manager: pod %s failed to evict %v", format.Pod(pod), err)
 	} else {
+		m.recorder.Eventf(pod, v1.EventTypeNormal, "KubeletEviction", "Successfully delete Pod as %s", msg)
```
Comment (yastij) [inline pkg/kubelet/eviction/eviction_manager.go]: Same, use past

**Review by yastij — CHANGES_REQUESTED** [review]

File: pkg/controller/util/node/controller_utils.go (hunk 1 above)
Comment (jiaxuanzhou) [reply pkg/controller/util/node/controller_utils.go]: modified

**Review by yastij — APPROVED** [review]

**Comment by yastij** [comment]
this LGTM overall, @deads2k to you for confirmation. also cc @gmarek for event side.

**Comment by deads2k** [comment]
> this LGTM overall, @deads2k to you for confirmation.

I'm fine with it.

/approve

**Comment by yastij** [comment]
@derekwaynecarr can you lgtm approve ?

**Comment by jiaxuanzhou** [comment]
@derekwaynecarr please help review and approve,thanks

File: pkg/controller/util/node/controller_utils.go [hunk 4]
```diff
@@ -97,6 +97,7 @@ func DeletePods(kubeClient clientset.Interface, recorder record.EventRecorder, n
 		if err := kubeClient.CoreV1().Pods(pod.Namespace).Delete(pod.Name, nil); err != nil {
 			return false, err
 		}
+		recorder.Eventf(&pod, v1.EventTypeNormal, "NodeControllerEviction", "Successfully deleted Pod %s from Node %s", pod.Name, nodeName)
```
Comment (gmarek) [inline pkg/controller/util/node/controller_utils.go]: We don't know if it's successful deletion at that point, we need to wait for ack from kubelet (and NC is not waiting for that). Additionally it violates Event style guide: https://github.com/kubernetes/community/blob/master/contributors/devel/event-style-guide.md

File: pkg/kubelet/eviction/eviction_manager.go [hunk 5]
```diff
@@ -576,18 +576,21 @@ func (m *managerImpl) evictPod(pod *v1.Pod, resourceName v1.ResourceName, evictM
...
-		Message: fmt.Sprintf(message, resourceName),
+		Message: msg,
 		Reason:  reason,
 	}
 	// record that we are evicting the pod
 	m.recorder.Eventf(pod, v1.EventTypeWarning, reason, evictMsg)
 	gracePeriod := int64(0)
 	err := m.killPodFunc(pod, status, &gracePeriod)
 	if err != nil {
+		m.recorder.Eventf(pod, v1.EventTypeWarning, "KubeletEviction", "Error deleting: %v", err)
```
Comment (gmarek) [inline pkg/kubelet/eviction/eviction_manager.go]: Same as above. Violates style guide. Events are not expected to mirror logs.

**Comment by gmarek** [comment]
This PR adds Events that violate second point of style guide https://github.com/kubernetes/community/blob/master/contributors/devel/event-style-guide.md. It should be closed.

File: pkg/kubelet/eviction/eviction_manager.go (hunk 5 above)
Comment (yastij) [reply pkg/kubelet/eviction/eviction_manager.go]: @gmarek - Oh I missed this one :) I thought that @jiaxuanzhou removed it as I linked the event-style guide.

File: pkg/kubelet/eviction/eviction_manager.go [hunk 6]
```diff
@@ -576,18 +576,21 @@ func (m *managerImpl) evictPod(pod *v1.Pod, resourceName v1.ResourceName, evictM
...
 		glog.Errorf("eviction manager: pod %s failed to evict %v", format.Pod(pod), err)
 	} else {
+		m.recorder.Eventf(pod, v1.EventTypeNormal, "KubeletEviction", "Successfully deleted Pod as %s", msg)
```
Comment (yastij) [inline pkg/kubelet/eviction/eviction_manager.go]: @gmarek - do we consider eviction something that could be noticed from resource alteration ?

**Comment by yastij** [comment]
@gmarek - I'm not against closing this one

**Comment by jiaxuanzhou** [comment]
@gmarek fine to close it

File: pkg/kubelet/eviction/eviction_manager.go (hunk 6 above)
Comment (gmarek) [reply pkg/kubelet/eviction/eviction_manager.go]: Yup - it's basically a time when Pod enters 'Terminated' (or something like that) phase.

**Pull request closed without merging**